eesp-ikev2.org

```diff
@@ -356,13 +355,17 @@ created by either peer and the key derivation for the in- and
 outbound EESP SAs of the Child SA are done as described in section
 2.17 of [[RFC7296]].
 
+
 If an SSKDF is selected as part of the proposal, instead of directly
 taking keys for the Sub SAs from KEYMAT, as described in section 2.17
-of [[RFC7296]], only one "root" key is taken for each EESP SA of the
-Child SA. Their length is determined by the key size of the
-negotiated SSKDF.  The root key for the EESP SA carrying data from
-the initiator to the responder is taken before that for the SA going
-from the responder to the initiator.
+of [[RFC7296]], a "root" key or a set of "root" keys per direction
+is derived for each EESP SA per direction of the Child SA. The number
+of keys depends on the cryptographic suite: AEAD transforms derive a
+single key per direction, while non-AEAD suites may derive separate
+keys for encryption and authentication. The input length of these
+keys, SK_sub, is determined by the key size of the negotiated SSKDF.
+The root key(s) for the initiator-to-responder direction are derivedi
+before those for the responder-to-initiator direction.
 
```

Comment:
That's not correct. We only get keys used for SSKDF here, not the individual keys for encryption/integrity algorithms. So this does not depend on the algorithms at all (except for the SSKDF key size). That difference is only relevant when applying SSKDF later to derive KEYMAT_sub using the key we got here.